lib/get-exports.js

```diff
@@ -57,23 +67,23 @@ async function getExports (url, context, parentLoad) {
   let source = parentCtx.source
   const format = parentCtx.format
 
-  // TODO support non-node/file urls somehow?
-  if (format === 'builtin') {
-    // Builtins don't give us the source property, so we're stuck
-    // just requiring it to get the exports.
-    return addDefault(Object.keys(require(url)))
-  }
-
   if (!source) {
-    // Sometimes source is retrieved by parentLoad, sometimes it isn't.
-    source = fs.readFileSync(fileURLToPath(url), 'utf8')
+    if (format === 'builtin') {
+      // Builtins don't give us the source property, so we're stuck
+      // just requiring it to get the exports.
+      return addDefault(Object.keys(require(url)))
```

Comment:
should we cache this? I guess should we cache the `fs.readFileSync` as well?

Reply:
Possibly, but this is only for built-ins. I'll benchmark it to see how slow it is.

I don't think we should cache the `readFileSync` calls because most files will only be hit once and we'll be left hanging onto the memory.

Reply:
In all the tests this specific require is only hit once so it's quite a niche code path.

There's another `Object.keys(require(id))` further down which is hit for every build-in module load from CJS code and the maximum I see is for `node:child_process` at about 1ms. These are probably worth caching because there's a limited number of possible built-ins.